### PR TITLE
adding DFE for Arabidopsis

### PR DESCRIPTION
**Pull request by chriscrsmith**
Addressing #1321 

Using estimated gamma shape and mean from https://doi.org/10.1038/s41467-018-05281-7. This study estimates dominance in addition to the DFE, however I think we decided to roll with the additive-only model because it is easier to implement.

Note that I plugged in [0, 1.0] for the proportions of neutral and selected loci.

Also I put a minus sign on the gamma mean which seems to be convention.

### CODE REVIEW

**Comment by petrelharp** [comment]
One thing (and the reason codecov is complaining): you need also to put
```
from . import dfes  # noqa somethingsomething
```
in `__init__.py`.

**Comment by petrelharp** [comment]
And: this DFE is estimated without dominance?

File: stdpopsim/catalog/AraTha/dfes.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    From Huber et al. (2018), https://doi.org/10.1038/s41467-018-05281-7.
+    Gamma parameters are based on the additive-only model for A. thaliana described in
+    Supplementary Table 4.
+    """
+    citations = [
+        stdpopsim.Citation(
+            author="Huber et al.",
+            year=2018,
+            doi="https://doi.org/10.1038/s41467-018-05281-7",
+            reasons="to be defined",  # include the dfe_model reason
```
Comment (petrelharp) [inline stdpopsim/catalog/AraTha/dfes.py:26]:
```suggestion
            reasons=stdpopsim.CiteReason.DFE,  # include the dfe_model reason
```

Comment (petrelharp) [reply stdpopsim/catalog/AraTha/dfes.py:26]: hm, I see you copied this from `catalog/HomSap/dfes.py`; could you fix this there, also?

**Comment by petrelharp** [comment]
> Note that I plugged in [0, 1.0] for the proportions of neutral and selected loci.

Is this right, or was it not clear from the paper?

**Comment by petrelharp** [comment]
> Also I put a minus sign on the gamma mean which seems to be convention.

Yes, good - that is how you specify deleterious `s` in SLiM.

File: stdpopsim/catalog/AraTha/dfes.py (hunk 1 above)
Comment (chriscrsmith) [reply stdpopsim/catalog/AraTha/dfes.py:26]: got it

**Comment by chriscrsmith** [comment]
> > Note that I plugged in [0, 1.0] for the proportions of neutral and selected loci.
> 
> Is this right, or was it not clear from the paper?

I am pretty sure this is right. In the supp mat table with the gamma parameters I used, it does not list any proportion neutral. And the main text makes it seem to me that neutral mutations were not included in most DFEs estimates.

**Comment by chriscrsmith** [comment]
> One thing (and the reason codecov is complaining): you need also to put

Doh. I must have gotten tired of waiting for the codecov to finish and I forgot to come back to this.

**Comment by chriscrsmith** [comment]
What does the comment "noqa ..." mean/do?

**Comment by grahamgower** [comment]
> noqa

This tells `flake8` to ignore the line. It can be used to ignore specific types of flake8 errors, e.g.
```
import foo  # noqa: F401
```
Means that flake8 won't produce the "imported but not used" error. The import may have desirable side effects, but flake8 doesn't know about this and just sees that the variable `foo` isn't used in the current scope.

File: stdpopsim/catalog/AraTha/__init__.py [hunk 2]
```diff
@@ -4,4 +4,5 @@
 from . import species  # noqa: F401
 from . import genetic_maps  # noqa: F401
 from . import demographic_models  # noqa: F401
-from . import annotations  # noqa: F401
+from . import dfes  # noqa: F401
+from . import annotations  # noqa: F401
```
Comment (petrelharp) [inline stdpopsim/catalog/AraTha/__init__.py:8]:
```suggestion
from . import annotations  # noqa: F401

```

**Comment by chriscrsmith** [comment]
> > noqa
> 
> This tells `flake8` to ignore the line. It can be used to ignore specific types of flake8 errors, e.g.
> 
> ```
> import foo  # noqa: F401
> ```
> 
> Means that flake8 won't produce the "imported but not used" error. The import may have desirable side effects, but flake8 doesn't know about this and just sees that the variable `foo` isn't used in the current scope.

Got it! Thanks

**Comment by petrelharp** [comment]
Awesome, thanks! I"ll merge this; could you open a QC issue for it, @chriscrsmith?

**Comment by chriscrsmith** [comment]
QC issue #1344

**Pull request merged**